react/src/components/BAIPropertyFilter.tsx

```diff
@@ -64,11 +64,17 @@ const BAIPropertyFilter: React.FC<BAIPropertyFilterProps> = ({
 }) => {
   const [search, setSearch] = useControllableValue({});
 
-  const [value, setValue] = useControllableValue<string | undefined>({
-    value: propValue,
-    defaultValue: defaultValue,
-    onChange: propOnChange,
-  });
+  const [value, setValue] = useControllableValue<string | undefined>(
+    _.omitBy(
+      {
+        value: propValue,
+        defaultValue: defaultValue,
+        onChange: propOnChange,
+      },
+      _.isEmpty,
```

Comment:
isEmpty() exists to manage empty value ( ex '').
but isn't control a controlled component even if an empty value is passed control as a value prop? not the same as undefined

Reply:
Oh, thank you for the feedback. It should be `isUndefined` instead of `isEmpty`. We need to decide how to handle the `undefined` value for the form control. I think it's better to work the same way as the React built-in `input` component. It should same between when the `value` is not provided or when the `value` is `undefined`.